src/commands/create/index.ts

```diff
@@ -0,0 +1,245 @@
+                message: `Config '${arg.id}' for interface '${interfaceEntry}': ${arg.description}`,
+                type: 'input',
+            };
+        },
+    };
+
+    appManifestInterfaces: AppManifestInterfaces = { interfaces: [] };
+
+    private _filterAvailableExamplesByLanguage(language: string) {
+        const filteredExamples = availableExamples.filter((example) => example.language === language);
+        if (!filteredExamples.length) {
+            throw new Error(`No example for your chosen language '${language}' available`);
+        }
+        return filteredExamples;
+    }
+
+    private async _runInteractiveMode(flags: any) {
+        flags.name = await ux.prompt(Create.prompts.name, { required: true });
+
+        let interactiveResponses: any = await inquirer.prompt([Create.prompts.language, Create.prompts.exampleQuestion]);
+
+        flags.language = interactiveResponses.language;
+        flags.example = interactiveResponses.exampleQuestion;
+
+        if (flags.example) {
+            availableExamples = this._filterAvailableExamplesByLanguage(flags.language);
+            interactiveResponses = await inquirer.prompt([Create.prompts.exampleUse]);
+        } else {
+            interactiveResponses = await inquirer.prompt([Create.prompts.interface]);
+        }
+        flags.example = interactiveResponses.exampleUse;
+        flags.interface = interactiveResponses.interface;
+
+        if (flags.interface && flags.interface.length > 0) {
+            await this._handleAdditionalInterfaceArgs(flags.interface);
+        }
+    }
+
+    private async _handleAdditionalInterfaceArgs(interfaces: string[]) {
+        for (const interfaceEntry of interfaces) {
+            const appManifestInterfaceEntry: AppManifestInterfaceEntry = { type: interfaceEntry, config: {} };
+            const interfaceObject = availableInterfaces.find(
+                (availableInterface: FunctionalInterfaceDescription) => availableInterface.value === interfaceEntry,
+            );
+            for (const arg of interfaceObject!.args) {
+                let interfaceArgResponse: any = {};
+                if (arg.required) {
+                    interfaceArgResponse = await inquirer.prompt([Create.prompts.additionalArg(arg, interfaceEntry)]);
+                    appManifestInterfaceEntry.config[arg.id] = interfaceArgResponse[arg.id];
+                } else {
+                    interfaceArgResponse[arg.id] = '';
+                }
+                if (!interfaceArgResponse[arg.id] && arg.default) {
+                    appManifestInterfaceEntry.config[arg.id] = arg.default;
+                    if (arg.type === 'object') {
+                        appManifestInterfaceEntry.config[arg.id] = JSON.parse(arg.default);
+                    }
+                }
+            }
+
+            this.appManifestInterfaces.interfaces.push(appManifestInterfaceEntry);
+        }
+    }
+
+    private _handlePackageIndex(packageIndex: PackageIndex) {
```

Comment:
Isn't this more of a "loading" step? How about calling this `_loadDataFromPackageIndex`